src/sentry/integrations/github/integration.py

```diff
@@ -399,57 +404,86 @@ def setup(self) -> None:
         )
 
 
-class OAuthLoginView(PipelineView):
-    def dispatch(self, request: Request, pipeline) -> HttpResponseBase:
-        self.determine_active_organization(request)
+class GitHubInstallationError(StrEnum):
+    INVALID_STATE = "Invalid state"
+    MISSING_TOKEN = "Missing access token"
+    MISSING_LOGIN = "Missing login info"
+    PENDING_DELETION = "GitHub installation pending deletion."
+    INSTALLATION_EXISTS = "Github installed on another Sentry organization."
+    USER_MISMATCH = "Authenticated user is not the same as who installed the app."
 
-        ghip = GitHubIdentityProvider()
-        github_client_id = ghip.get_oauth_client_id()
-        github_client_secret = ghip.get_oauth_client_secret()
 
-        installation_id = request.GET.get("installation_id")
-        if installation_id:
-            pipeline.bind_state("installation_id", installation_id)
+def record_event(event: IntegrationPipelineViewType):
+    return IntegrationPipelineViewEvent(event, "source_code", GitHubIntegrationProvider.key)
 
-        if not request.GET.get("state"):
-            state = pipeline.signature
 
-            redirect_uri = absolute_uri(
-                reverse("sentry-extension-setup", kwargs={"provider_id": "github"})
-            )
-            return self.redirect(
-                f"{ghip.get_oauth_authorize_url()}?client_id={github_client_id}&state={state}&redirect_uri={redirect_uri}"
+class OAuthLoginView(PipelineView):
+    def dispatch(self, request: Request, pipeline) -> HttpResponseBase:
+        with record_event(IntegrationPipelineViewType.OAUTH_LOGIN).capture() as lifecycle:
+            self.determine_active_organization(request)
+            lifecycle.add_extra(
+                "organization_id",
+                self.active_organization.organization.id if self.active_organization else None,
             )
```

Comment:
Not super applicable for this PR I guess, but would it make sense for us to fold extra data like this into the `record_event(...).capture()` call? Are there cases when we'd want to conditionally add data like this vs just specifying it once at the beginning of the context manager block?

Reply:
this data is not present at the beginning of the block so i'm adding it in here